devito/data/allocators.py

```diff
@@ -92,8 +92,8 @@ def initialize(cls):
         return
 
     def alloc(self, shape, dtype, padding=0):
-        datasize = int(reduce(mul, shape))
-        ctype = dtype_to_ctype(dtype)
+        ctype, c_scale = dtype_alloc_ctype(dtype)
```

Comment:
nitpicking , I'd say "scale" or "cscale" for homogeneity

Reply:
and most importantly why don't you just compute `datasize` inside `dtype_alloc_ctype` directly ? in the only other place in which `dtype_alloc_ctype` is used, you are computing a `datasize` too, so I'd move everything into that function, and perhaps rename it simply as `infer_datasize`?